### PR TITLE
Massive rewrite of wellness to be more flexible

### PR DESCRIPTION
**Pull request by warmwaffles**
- Services can now be Procs or lambdas
- Super flexible configuring for wellness
- Switched to minitest as the test framework
- Removed rspec dependency
- Removed simplecov dependency

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,58 +97,120 @@ use(Wellness::Middleware, system)
...
+        'details' => details
       }
-    else
+    end
+
+    def unhealthy(details={})
       {
-        'status': 'UNHEALTHY'
+        'status' => 'UNHEALTHY',
+        'details' => {
```
Comment (supernullset) [inline README.md]: You may want to do `'details' => details` here just for consistency with your above example

Comment (warmwaffles) [inline README.md]: oops yea, that is a good point.

File: lib/wellness/simple_report.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    end
+
+    def call
+      non_criticals = []
+      criticals = []
+
+      @system.services.each do |name, service|
+        result = service.call['status']
+
+        if service.critical?
```
Comment (supernullset) [inline lib/wellness/simple_report.rb:22]: I feel like everything from here to L41 could be combined. The status checking and such could be condensed into the initial iteration over services. I dont think this is at all critical right nowl; everything looks like it works just fine.

**Comment by supernullset** [comment]
:+1:

File: lib/wellness/simple_report.rb (hunk 2 above)
Comment (warmwaffles) [inline lib/wellness/simple_report.rb:22]: This part of the library is not exposed to the consumer, so we have the flexibility to do what we want as long as all services and details are callable.

**Pull request merged**